src/test/java/org/isf/vaccine/test/Tests.java

```diff
@@ -134,13 +141,181 @@ public void testIoIsCodePresent() throws Exception {
 	}
 
 	@Test
-	public void testFindVaccine() throws Exception {
+	public void testIoFindVaccine() throws Exception {
 		String code = _setupTestVaccine(false);
 		Vaccine result = vaccineIoOperation.findVaccine(code);
 		assertThat(result).isNotNull();
 		assertThat(result.getCode()).isEqualTo(code);
 	}
 
+	@Test
+	public void testIoFindVaccineNull() throws Exception {
+		assertThatThrownBy(() -> vaccineIoOperation.findVaccine(null))
+				.isInstanceOf(RuntimeException.class);
+	}
+
+	@Test
+	public void testMgrGetVaccineShouldFindByTypeCode() throws Exception {
+		String code = _setupTestVaccine(false);
+		Vaccine foundVaccine = vaccineBrowserManager.findVaccine(code);
+		ArrayList<Vaccine> vaccines = vaccineBrowserManager.getVaccine(foundVaccine.getVaccineType().getCode());
+		assertThat(vaccines.get(vaccines.size() - 1).getDescription()).isEqualTo(foundVaccine.getDescription());
+	}
+
+	@Test
+	public void testMgrGetVaccineShouldFindAllVaccinesWhenNoCodeProvided() throws Exception {
+		String code = _setupTestVaccine(false);
+		Vaccine foundVaccine = vaccineBrowserManager.findVaccine(code);
```

Comment:
this line is not being used  `foundVaccine`

Reply:
same in line 96

Reply:
@dbmalkovsky +1